tap_mssql/sync_strategies/log_based.py

```diff
@@ -1,396 +1,405 @@
+
+    # store the state of the table lsn's after the initial load ready for the next CDC run
+    singer.write_message(singer.StateMessage(value=copy.deepcopy(state)))
+
+    # clear max pk value and last pk fetched upon successful sync
+    singer.clear_bookmark(state, catalog_entry.tap_stream_id, "max_pk_values")
+    singer.clear_bookmark(state, catalog_entry.tap_stream_id, "last_pk_fetched")
+
+    singer.write_message(activate_version_message)
+
+
+def sync_table(mssql_conn, config, catalog_entry, state, columns, stream_version):
+    mssql_conn = MSSQLConnection(config)
+    common.whitelist_bookmark_keys(
+        generate_bookmark_keys(catalog_entry), catalog_entry.tap_stream_id, state
+    )
+
+    # Add additional keys to the columns
+    extended_columns = columns + [
+        "_sdc_operation_type",
+        "_sdc_lsn_commit_timestamp",
+        "_sdc_lsn_deleted_at",
+        "_sdc_lsn_value",
+        "_sdc_lsn_seq_value",
+        "_sdc_lsn_operation",
+    ]
+
+    bookmark = state.get("bookmarks", {}).get(catalog_entry.tap_stream_id, {})
+    version_exists = True if "version" in bookmark else False
+
+    initial_full_table_complete = singer.get_bookmark(
+        state, catalog_entry.tap_stream_id, "initial_full_table_complete"
+    )
+
+    state_version = singer.get_bookmark(state, catalog_entry.tap_stream_id, "version")
+
+    activate_version_message = singer.ActivateVersionMessage(
+        stream=catalog_entry.stream, version=stream_version
+    )
+
+    # For the initial replication, emit an ACTIVATE_VERSION message
+    # at the beginning so the records show up right away.
+    if not initial_full_table_complete and not (version_exists and state_version is None):
+        singer.write_message(activate_version_message)
+
+    with connect_with_backoff(mssql_conn) as open_conn:
+        with open_conn.cursor() as cur:
+
+            state_last_lsn = singer.get_bookmark(state, catalog_entry.tap_stream_id, "lsn")
+
+            escaped_columns = map(lambda c: common.prepare_columns_sql(catalog_entry, c), columns)
+            table_name = catalog_entry.table
+            schema_name = common.get_database_name(catalog_entry)
+            cdc_table = schema_name + "_" + table_name
+            escape_table_name = common.escape(catalog_entry.table)
+            escaped_schema_name = common.escape(schema_name)
+            escaped_cdc_table = common.escape(cdc_table)
+
+            if not verify_change_data_capture_table(mssql_conn, escaped_schema_name, escape_table_name):
+                raise Exception(
+                    (
+                        "Error {}.{}: does not have change data capture enabled. "
+                        "Call EXEC sys.sp_cdc_enable_table with relevant parameters to enable CDC."
+                    ).format(escaped_schema_name, escape_table_name)
+                )
+
+            lsn_range = get_lsn_available_range(mssql_conn, escaped_cdc_table)
+
```

Comment:
This is good, I can see why escaped table, e.g. "Thishasa$dollar" is used in the get lsn query, but that needs to be 'un-escaped' when built as part of the CDC function name